### PR TITLE
[GOBBLIN-2113] Process Heartbeat DagAction CDC messages with empty FlowExecutionId str

### PR DESCRIPTION
**Pull request by umustafi**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-2113

### Description
- [X] Here are some details about my PR, including screenshots (if applicable):
After changing `flowExecutionId` from a Str to a long in a previous PR, we encounter `NumberFormatException` in the `DagActionStoreChangeMonitor` when processing HB events. This ends up killing the `HighLevelConsumer` queues for the hosts that receive the HB events in their partition. 
```
{{Encountered exception while processing record so stopping queue processing. Record: LiKafka10ConsumerRecord(consumerRecord=ConsumerRecord(topic = ds_mysql_makto-db-152_prod_SHARED_GOBBLIN_DAG_ACTION_STORE_20221208211255, partition = 0, leaderEpoch = null, offset = 905733, NoTimestampType = -1, serialized key size = -1, serialized value size = -1, headers = RecordHeaders(headers = [], isReadOnly = false), key = , value = {"changeEventIdentifier":
{"key": "", "txId": "", "produceTimestampMillis": 1721078158347, "operationType": "HEARTBEAT"}
, "flowGroup": "", "flowName": "", "flowExecutionId": "", "jobName": "", "dagAction": null})) Exception: java.lang.NumberFormatException: For input string: ""}}
```

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Updates the following unit test for Heartbeat (HB) events which was using valid flow names, groups, and flowExecutionId for HB event that did not accurately reflect a HB event received from CDC. After updating its values the test failed locally 

```
Gradle suite > Gradle test > org.apache.gobblin.runtime.DagActionStoreChangeMonitorTest > testProcessMessageWithHeartbeatAndNullDagAction FAILED
    java.lang.NumberFormatException: For input string: ""
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.lang.Long.parseLong(Long.java:601)
        at java.lang.Long.parseLong(Long.java:631)
        at org.apache.gobblin.service.monitoring.DagActionStoreChangeMonitor.processMessage(DagActionStoreChangeMonitor.java:203)
        at org.apache.gobblin.runtime.DagActionStoreChangeMonitorTest$MockDagActionStoreChangeMonitor.processMessageForTest(DagActionStoreChangeMonitorTest.java:97)
        at org.apache.gobblin.runtime.DagActionStoreChangeMonitorTest.testProcessMessageWithHeartbeatAndNullDagAction(DagActionStoreChangeMonitorTest.java:139)
Failed tests:
[org.apache.gobblin.runtime.DagActionStoreChangeMonitorTest::testProcessMessageWithHeartbeatAndNullDagAction
```

It passes after the update. 

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/runtime/DagActionStoreChangeMonitorTest.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public void testProcessMessageWithHeartbeatAndNullDagAction() throws SpecNotFoun
   @Test (dependsOnMethods = "testProcessMessageWithHeartbeatAndNullDagAction")
   public void testProcessMessageWithHeartbeatAndFlowInfo() throws SpecNotFoundException {
     Kafka09ConsumerClient.Kafka09ConsumerRecord consumerRecord =
-        wrapDagActionStoreChangeEvent(OperationType.HEARTBEAT, FLOW_GROUP, FLOW_NAME, FLOW_EXECUTION_ID, DagActionValue.RESUME);
+        wrapDagActionStoreChangeEvent(OperationType.HEARTBEAT, "", "", "", DagActionValue.RESUME);
```
Comment (arjun4084346) [inline gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/runtime/DagActionStoreChangeMonitorTest.java:153]: is this a real case? can a heartbeat message not have flow name and have dag action?

**Review by arjun4084346 — APPROVED** [review]

File: gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/runtime/DagActionStoreChangeMonitorTest.java (hunk 1 above)
Comment (umustafi) [reply gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/runtime/DagActionStoreChangeMonitorTest.java]: Not a real case but wanted to check some edge case behavior

**Comment by umustafi** [comment]
checks on my branch https://github.com/umustafi/gobblin/pull/34

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java [hunk 2]
```diff
@@ -112,20 +112,26 @@ public Optional<Dag<JobExecutionPlan>> createExecutionPlanIfValid(FlowSpec flowS
 
     TimingEvent flowCompilationTimer = new TimingEvent(this.eventSubmitter, TimingEvent.FlowTimings.FLOW_COMPILED);
     Map<String, String> flowMetadata = TimingEventUtils.getFlowMetadata(flowSpec);
-    Optional<Dag<JobExecutionPlan>> jobExecutionPlanDagOptional =
-        validateAndHandleConcurrentExecution(flowConfig, flowSpec, flowGroup, flowName, flowMetadata);
 
-    if (!jobExecutionPlanDagOptional.isPresent()) {
-      return Optional.absent();
-    }
+    try {
```
Comment (Will-Lo) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java:116]: Do we still want the try/catch here?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java [hunk 3]
```diff
@@ -112,20 +112,26 @@ public Optional<Dag<JobExecutionPlan>> createExecutionPlanIfValid(FlowSpec flowS
...
+        populateFlowCompilationFailedEventMessage(eventSubmitter, flowSpec, flowMetadata);
+        return Optional.absent();
+      }
 
-    flowCompilationTimer.stop(flowMetadata);
-    return jobExecutionPlanDagOptional;
+      flowCompilationTimer.stop(flowMetadata);
+      return jobExecutionPlanDagOptional;
+    } catch (IOException e) {
+      log.error("Encountered exception when attempting to compile and perform checks for flow: {}", flowSpec);
```
Comment (Will-Lo) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java:132]: If you want to log this out maybe only print out the flowspec IDs

**Review by Will-Lo — APPROVED** [review]
